### PR TITLE
Attestation_API_Proposal.md: add flag to oe_get_evidence() to indicate if evidence will be wrapped

### PR DESCRIPTION
**Pull request by shnwc**
Add flag to API function `oe_get_evidence()` to indicate if the output evidence will be wrapped with an attestation header that embeds the format ID. This is in response to comment https://github.com/openenclave/openenclave/pull/3198#issuecomment-652675145

### CODE REVIEW

**Comment by shnwc** [comment]
The latest commit has the function `oe_get_evidence()` taking a bit-wise flags parameter, with one bit indicating if evidence will be wrapped with an attestation header.

**Comment by shnwc** [comment]
The latest commit incorporated all the comments, and is ready for merge.

File: docs/DesignDocs/Attestation_API_Proposal.md [hunk 1]
```diff
@@ -624,29 +600,52 @@ by `oe_get_report()`.
...
-will be raw EPID quote. Raw EPID quotes are what existing SGX EPID-based
-solutions have been using, this design will help maintain backward
+used as the SGX quote `report_data` directly. This design is consistent with
+existing SGX EPID-based solutions behavior, and will help maintain backward
 compatibility.
 
+The table below shows the structure of the evidence data for all the
+SGX format IDs, as generated by an attester plugin or verified by a verifier
+plugin. For every listed format, `oe_get_evidence()` can wrap with
+an attestation header when its `wrap_with_header` parameter is set to `true`.
```
Comment (CodeMonkeyLeet) [inline docs/DesignDocs/Attestation_API_Proposal.md:629]: Note, this is inconsistent with the function signature earlier in the doc now.

Comment (shnwc) [reply docs/DesignDocs/Attestation_API_Proposal.md]: Thanks! Fixed in the latest commit.

**Pull request merged**